apps/web/components/booking/BookingListItem.tsx

```diff
@@ -654,45 +672,255 @@ const FirstAttendee = ({
 type AttendeeProps = {
   name?: string;
   email: string;
+  id: number;
+  noShow: boolean;
+};
+
+type NoShowProps = {
+  bookingUid: string;
+  isPast: boolean;
 };
 
-const Attendee = ({ email, name }: AttendeeProps) => {
+const Attendee = (attendeeProps: AttendeeProps & NoShowProps) => {
+  const { email, name, bookingUid, isPast, noShow: noShowAttendee } = attendeeProps;
+  const { t } = useLocale();
+
+  const [noShow, setNoShow] = useState(noShowAttendee);
+  const [openDropdown, setOpenDropdown] = useState(false);
+
+  const noShowMutation = trpc.viewer.public.noShow.useMutation({
+    onSuccess: async () => {
+      noShow
+        ? showToast(
+            t("x_marked_as_no_show", {
+              x: email,
+            }),
+            "success"
+          )
+        : showToast(
+            t("x_unmarked_as_no_show", {
+              x: email,
+            }),
+            "success"
+          );
+    },
+    onError: (err) => {
+      showToast(err.message, "error");
+    },
+  });
+
+  function toggleNoShow({
+    attendee,
+    bookingUid,
+  }: {
+    attendee: { email: string; noShow: boolean };
+    bookingUid: string;
+  }) {
+    noShowMutation.mutate({ bookingUid, attendees: [attendee] });
+    setNoShow(!noShow);
```

Comment:
Optimistic update. We can also do it async and wait for the result here, but since it is a simple update call that pretty much never has to change, this may or may not be needed. Open to either.

Reply:
Yeah lets stick with optimistic in places that make sense.